src/main/java/org/opensearch/performanceanalyzer/reader/SearchBackPressureMetricsSnapShot.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.reader;
+
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.BatchBindStep;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.opensearch.performanceanalyzer.commons.metrics.AllMetrics;
+
+public class SearchBackPressureMetricsSnapShot implements Removable {
```

Comment:
Does this table require any `fetchAggregated`, `groupBy` or `fetchBySpecificColumn` ?

Executing a `fetchAll` and then running aggregation on the Java layer is expensive, instead leverage the sqlite functionality to aggregate/filter/groupBy the data and plugin the resultset directly into the RCA/API response.

Reply:
In every basic timeunit (5s) collector populate the shared memory folder and only 1 sqlitedb table get created. Also, since this table do not require any `fetchAggregated`, `groupBy` or `fetchBySpecificColumn`, thus, only fetchAll would suffice.

Reply:
The cluster/node level RCA would consume these data in the snapshots and determine whether the search back pressure service has cancelled too much/ too less requests, by comparing with predefined threshold.

Reply:
Resolved in latest commit